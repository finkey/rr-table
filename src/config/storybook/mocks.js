export const SAMPLE_TEXT = 'sample text';

export const LONG_TEXT = 'Lorem ipsum dolor amet jianbing vegan art party health goth PBR&B tumeric yuccie everyday carry farm-to-table air plant XOXO meh kickstarter. Fingerstache kale chips lumbersexual flexitarian jean shorts normcore';

export const breakpoints = [500, 700, 900, 1000, 1100];

export const priorities = [1, 2, undefined, 3, 4];

export const colWidths = [2, 1.5, undefined, 0.8, 2];

export const items = [
  'sample text - 1',
  'sample text hahah haha haha - 2',
  'sample text Bonjour - 3',
  'sample text - 4',
  'sample text  Bla bla bla dsfjk sldkifh qshdfm shdsmldhf shd  - 5',
  '',
  'sample text - 7',
];

export const titles = ['Name', 'Surname', 'Age', 'Job', 'Sex', 'Animaux'];

export const keys = ['name', 'surname', 'info.age', 'company.job', 'info.sex', 'pets'];

export const list = [
  {
    name: 'Francis',
    surname: 'Denis',
    info: {
      age: 33,
      sex: 'male',
    },
    company: {
      job: 'artist',
    },
    id: '1',
    pets: ['dog', 'cat'],
  },
  {
    name: 'Ingals Bla bla bla',
    surname:
      'Charles Le plus beau et le plus gros et le plus sympas, il est rigolo hahah hahah ahahah ahahah',
    info: {
      age: 47,
      sex: 'male',
    },
    company: {
      job: 'teacher',
    },

    id: '2',
  },
  {
    name: 'Dumond',
    surname:
      'Charlotte Lorem ipsum dolor amet vape cornhole lyft fanny pack, before they sold out subway tile tilde wolf distillery shoreditch. Shabby chic 8-bit brooklyn brunch. Kinfolk live-edge tbh succulents, portland art party cloud bread venmo gentrify hashtag brooklyn yuccie vegan twee fingerstache. Gochujang salvia small batch cold-pressed meggings art party gluten-free, celiac vice fam flexitarian. Franzen tote bag shabby chic synth, tofu ennui single-origin coffee kale chips thundercats affogato lo-fi twee keffiyeh palo santo fam. IPhone yuccie polaroid taxidermy food truck letterpress mlkshk. 8-bit deep v thundercats, XOXO bicycle rights freegan aesthetic meh 3 wolf moon.',
    info: {
      age: 53,
      sex: 'female',
    },
    company: {
      job: 'astraunot',
    },

    id: '3',
    pets: ['fish', 'cat'],
  },
  {
    name: 'Applencourt',
    surname: 'Francoise',
    info: {
      age: 23,
      sex: 'female',
    },
    company: {
      job: 'singer',
    },

    id: '4',
  },
  {
    name: 'Dupond',
    surname: 'Herver',
    info: {
      age: 33,
      sex: 'male',
    },
    company: {
      job: 'chef',
    },

    id: '5',
    pets: ['dog'],
  },
  {
    name: 'Pierre',
    surname: 'Arthur',
    info: {
      age: 63,
      sex: 'male',
    },
    company: {
      job: '',
    },

    id: '6',
  },
  {
    name: 'Dicaprio',
    surname: 'Jose',
    info: {
      age: 34,
      sex: 'male',
    },
    company: {
      job: 'actor',
    },

    id: '7',
  },
  {
    name: 'James',
    surname: 'Lebron',
    info: {
      age: 19,
      sex: 'male',
    },
    company: {
      job: 'footballer',
    },

    id: '8',
  },
];
