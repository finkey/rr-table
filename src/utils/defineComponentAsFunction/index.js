import { isValidElement } from 'react';

const defineComp = (component, defaultComponent) => {
  if (isValidElement(component)) {
    return () => component;
  }

  if (typeof component === 'function') {
    return component;
  }

  if (isValidElement(defaultComponent)) {
    return () => defaultComponent;
  }

  if (typeof defaultComponent === 'function') {
    return defaultComponent;
  }
};

export default defineComp;
