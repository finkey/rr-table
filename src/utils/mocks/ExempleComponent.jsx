import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 100%;
  border: 2px solid #9fa8da;
  padding: 2rem;
  color: ${({ color }) => color};
`;

const ExempleComponent = ({ color, text }) => <Wrapper color={color}>{text}</Wrapper>;

ExempleComponent.propTypes = { text: PropTypes.string, color: PropTypes.string };

export default ExempleComponent;
