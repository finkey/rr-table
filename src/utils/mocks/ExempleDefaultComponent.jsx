import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 100%;
  border: 2px solid #26a69a;
  padding: 1rem;
  color: ${({ color }) => color};
`;

const ExempleDefaultComponent = ({ color, text }) => <Wrapper color={color}>{text}</Wrapper>;

ExempleDefaultComponent.propTypes = { text: PropTypes.string, color: PropTypes.string };

export default ExempleDefaultComponent;
