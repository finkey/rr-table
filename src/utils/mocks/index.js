export { default as ExempleComponent } from './ExempleComponent';
export { default as ExempleDefaultComponent } from './ExempleDefaultComponent';
