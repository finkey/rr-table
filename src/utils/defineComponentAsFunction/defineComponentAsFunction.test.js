import React from 'react';
// import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { ExempleComponent, ExempleDefaultComponent } from '../mocks';
import defineComponentAsFunction from './index';

describe('defineComponentAsFunction', () => {
  const passedProps = { text: 'hello', color: 'papayawhip' };

  it('should return a function that returns component when component isValidElement', () => {
    const component = <ExempleComponent {...passedProps} />;
    const definedComponent = defineComponentAsFunction(component);
    const Comp = renderer.create(definedComponent());
    const tree = Comp.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return component when component is a function', () => {
    const component = ExempleComponent;
    const definedComponent = defineComponentAsFunction(component);
    const Comp = renderer.create(definedComponent({ ...passedProps }));
    const tree = Comp.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return a function that returns defaultComponent when defaultComponent isValidElement and component is undefined', () => {
    const component = undefined;
    const defaultComp = <ExempleDefaultComponent {...passedProps} />;
    const definedComponent = defineComponentAsFunction(component, defaultComp);
    const Comp = renderer.create(definedComponent());
    const tree = Comp.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return defaultComponent when defaultComponent is a function and component is undefined', () => {
    const component = undefined;
    const defaultComp = ExempleDefaultComponent;
    const definedComponent = defineComponentAsFunction(component, defaultComp);
    const Comp = renderer.create(definedComponent({ ...passedProps }));
    const tree = Comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
