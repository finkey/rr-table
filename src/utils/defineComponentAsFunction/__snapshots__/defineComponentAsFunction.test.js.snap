// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defineComponentAsFunction should return a function that returns component when component isValidElement 1`] = `
.c0 {
  width: 100%;
  border: 2px solid #9fa8da;
  padding: 2rem;
  color: papayawhip;
}

<div
  className="c0"
  color="papayawhip"
>
  hello
</div>
`;

exports[`defineComponentAsFunction should return a function that returns defaultComponent when defaultComponent isValidElement and component is undefined 1`] = `
.c0 {
  width: 100%;
  border: 2px solid #26a69a;
  padding: 1rem;
  color: papayawhip;
}

<div
  className="c0"
  color="papayawhip"
>
  hello
</div>
`;

exports[`defineComponentAsFunction should return component when component is a function 1`] = `
.c0 {
  width: 100%;
  border: 2px solid #9fa8da;
  padding: 2rem;
  color: papayawhip;
}

<div
  className="c0"
  color="papayawhip"
>
  hello
</div>
`;

exports[`defineComponentAsFunction should return defaultComponent when defaultComponent is a function and component is undefined 1`] = `
.c0 {
  width: 100%;
  border: 2px solid #26a69a;
  padding: 1rem;
  color: papayawhip;
}

<div
  className="c0"
  color="papayawhip"
>
  hello
</div>
`;
